src/network_analysis.jl

```diff
@@ -387,32 +427,43 @@ rcs,incidencemat = reactioncomplexes(sir)
 ```
 """
 function deficiency(rn::ReactionSystem)
+    # Precomputes required information. `conservationlaws` caches the conservation laws in `rn`.
     nps = get_networkproperties(rn)
     conservationlaws(rn)
     r = nps.rank
     ig = incidencematgraph(rn)
     lc = linkageclasses(rn)
+
+    # Computes deficiency using its formula. Caches and returns it as output.
     nps.deficiency = Graphs.nv(ig) - length(lc) - r
     nps.deficiency
 end
 
-# Used in the subsequent function.
+# For a linkage class (set of reaction complexes that form an isolated sub-graph), and some
+# additional information of the full network, find the reactions, species, and parameters 
+# that constitute the corresponding sub-reaction network.
 function subnetworkmapping(linkageclass, allrxs, complextorxsmap, p)
+    # Finds the reactions that are part of teh sub-reaction network.
     rxinds = sort!(collect(Set(rxidx for rcidx in linkageclass
                                for rxidx in complextorxsmap[rcidx])))
-    rxs = allrxs[rxinds]
-    specset = Set(s for rx in rxs for s in rx.substrates if !isconstant(s))
-    for rx in rxs
+    newrxs = allrxs[rxinds]
+
+    # Find the species that are part of the sub-reaction network.
+    specset = Set(s for rx in newrxs for s in rx.substrates if !isconstant(s))
```

Comment:
Mark as new line. It is not, it has simply been pushed down one step and uses `newrxs` instead of `rxs`.